components/configuration-mgt/org.wso2.carbon.identity.configuration.mgt.core/src/main/java/org/wso2/carbon/identity/configuration/mgt/core/dao/impl/CachedBackedConfigurationDAO.java

```diff
@@ -363,6 +365,43 @@ private Resource getResourceFromCacheByName(String resourceName, String tenantDo
         return null;
     }
 
+    private void removeResourceFromCache(String resourceId, int tenantId) throws ConfigurationManagementException {
+
+        Resource resource = getResourceFromCacheById(resourceId, tenantId);
+        if (resource == null) {
+            if (log.isDebugEnabled()) {
+                log.debug("No resource found in the cache with resourceId: " + resourceId);
+            }
+            return;
+        }
+        String resourceName = resource.getResourceName();
+        removeResourceFromCache(tenantId, resourceId, resourceName);
+    }
+
+    private void removeResourceFromCache(int tenantId, String resourceId, String resourceName) {
+
+        String tenantDomain = PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+        try {
+            PrivilegedCarbonContext.startTenantFlow();
+            PrivilegedCarbonContext.getThreadLocalCarbonContext()
+                    .setTenantDomain(MultitenantConstants.SUPER_TENANT_DOMAIN_NAME);
+            PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantId(tenantId);
+            ResourceByIdCacheKey resourceByIdCacheKey = new ResourceByIdCacheKey(resourceId, tenantDomain);
+            ResourceByNameCacheKey resourceByNameCacheKey = new ResourceByNameCacheKey(resourceName, tenantDomain);
+            ResourceCacheEntry resourceCacheEntry = resourceByIdCache.getValueFromCache(resourceByIdCacheKey);
+            if (resourceCacheEntry == null) {
+                if (log.isDebugEnabled()) {
+                    log.debug("No cache entry found for resource: " + resourceId);
+                }
+                return;
```

Comment:
don't we need to remove from resourceByNameCache?

Reply:
Fixed in afaf1aa2548752b885a92a7ac5e783a8f7e02397